### DISCUSSION TITLE
Custom Configuration (strategies): Is there an undocumented implicit ordering or is the ordering explicit?

### DISCUSSION THREAD

**Discussion by fourpastmidnight** [GitTools/GitVersion#4248] [Q&A] [answered]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have searched [issues](https://github.com/GitTools/GitVersion/issues) to ensure it has not already been reported

### GitVersion package

GitVersion.Tool

### GitVersion version

6.0.2

### Operating system

Windows

### What are you seeing?

This is a documentation question. I'm going to create a custom configuration. I have looked at the documentation and the default configurations for each of the GitFlow, GitHubFlow, Mainline, and TrunkDevelopment models of versioning. Each defines a `strategies` array as part of the configuration. The only thing is, it's not documented if this is an unordered list (with some baked-in implicit ordering performed by GitVersion itself) or if this list is, in fact, explicitly ordered. If it is explicitly ordered, it _appears_ that it may be that the first one listed (which is usually `Fallback`) is the least prioritized, so that would mean the last one listed has the highest precedence. But this is all an assumption based on no documentation.

So, when creating a custom configuration and setting the versioning strategies to be used, is the ordering of the list important? Can it be controlled?

### What is expected?

I expect that there is documentation for the `strategies` property of the GitVersion YAML configuration file, and that such documentation would state whether the list is an explicitly ordered list, or whether the list is unordered with GitVersion implementing an implicit ordering of the available strategies.

### Steps to Reproduce

N/A

**Comment by HHobeck** [comment, marked as answer]
The order of the defined strategies doesn't matter. The fallback strategy only applies if no other version strategies in the list return a base version.

**Reply by fourpastmidnight** [reply]
Thank you!
